sqlstore/src/test/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateSampleDaoTest.java

```diff
@@ -274,27 +274,65 @@ public void countSamplesByEmptySearch() throws IOException {
   }
 
   @Test
-  public void testGetIdentityByExternalName() throws IOException {
-    List<SampleIdentity> identity = (List<SampleIdentity>) dao.getIdentitiesByExternalNameOrAlias("EXT15");
-    assertTrue(identity.get(0).getExternalName().contains("EXT15"));
+  public void testGetIdentityByPartialMatchExternalName() throws IOException {
+    List<SampleIdentity> identity = (List<SampleIdentity>) dao.getIdentitiesByExternalNameOrAliasPartialMatch("EXT1");
+    assertTrue(identity.get(0).getExternalName().contains("EXT1"));
+  }
+
+  @Test
+  public void testGetIdentityByExactMatchExternalNameWithNonExactMatch() throws IOException {
+    String query = "EXT1";
+    List<SampleIdentity> exactMatches = (List<SampleIdentity>) dao.getIdentitiesByExactExternalName(query);
+    assertTrue(exactMatches.isEmpty());
+  }
+
+  @Test
+  public void testGetidentityByExactMatchExternalNameWithExactMatch() throws IOException {
+    String query = "EXT15";
+    List<SampleIdentity> exactMatches = (List<SampleIdentity>) dao.getIdentitiesByExactExternalName(query);
+    assertFalse(exactMatches.isEmpty());
+  }
+
+  @Test
+  public void testGetIdentityByExactMatchExternalNameWithEmptyString() throws IOException {
+    String query = "";
+    Collection<SampleIdentity> exactMatches = dao.getIdentitiesByExactExternalName(query);
+    assertTrue(exactMatches.isEmpty());
+  }
+
+  @Test
+  public void testGetIdentityByExternalNameAndProjectExactMatch() throws IOException {
+    String externalName = "EXT15";
+    long projectId = 1;
+    Collection<SampleIdentity> exactMatches = dao.getIdentitiesByExactExternalNameAndProject(externalName, projectId);
+    assertFalse(exactMatches.isEmpty());
+    assertEquals(exactMatches.size(), 1);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Swap these 2 arguments so they are in the correct order: expected value, actual value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3415)